include/deal.II/physics/notation.h

```diff
@@ -0,0 +1,1513 @@
+
+
+      /**
+       * Convert a compressed matrix to its equivalent scalar value.
+       */
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Number                   &s,
+                      const FullMatrix<Number> &mtrx);
+
+
+      /**
+       * Convert a compressed matrix to its equivalent rank-0 tensor.
+       */
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Tensor<0,dim,Number>     &s,
+                      const FullMatrix<Number> &mtrx);
+
+
+      /**
+       * Convert a compressed matrix to its equivalent rank-1 tensor.
+       */
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Tensor<1,dim,Number>     &v,
+                      const FullMatrix<Number> &mtrx);
+
+
+      /**
+       * Convert a compressed matrix to its equivalent rank-2 tensor.
+       */
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Tensor<2,dim,Number>     &t,
+                      const FullMatrix<Number> &mtrx);
+
+
+      /**
+       * Convert a compressed matrix to its equivalent rank-2 symmetric tensor.
+       */
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (SymmetricTensor<2,dim,Number> &st,
+                      const FullMatrix<Number>      &mtrx);
+
+
+      /**
+       * Convert a compressed matrix to its equivalent rank-3 tensor.
+       *
+       * @note Based on the size of the matrix @p mtrx, some of the
+       * components of @p t may be interpreted as having symmetric
+       * counterparts. This is the reverse of the operation explained
+       * in the documentation of the counterpart unroll_to_matrix()
+       * function.
+       */
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Tensor<3,dim,Number> &t,
+                      const FullMatrix<Number> &mtrx);
+
+
+      /**
+       * Convert a compressed matrix to its equivalent rank-4 tensor.
+       */
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (Tensor<4,dim,Number> &t,
+                      const FullMatrix<Number> &mtrx);
+
+
+      /**
+       * Convert a compressed matrix to its equivalent rank-4 symmetric tensor.
+       */
+      template<int dim, typename Number>
+      void
+      roll_to_tensor (SymmetricTensor<4,dim,Number> &st,
+                      const FullMatrix<Number> &mtrx);
+
+
+      /**
+       * A generic helper function that will convert a compressed vector
+       * to its equivalent @p TensorType.
+       */
+      template<int dim, typename TensorType, typename Number>
+      TensorType
+      roll_to_tensor (const Vector<Number> &vec);
+
+
+      /**
+       * A generic helper function that will convert a compressed matrix
+       * to its equivalent @p TensorType.
+       */
+      template<int dim, typename TensorType, typename Number>
+      TensorType
+      roll_to_tensor (const FullMatrix<Number> &vec);
+//@}
+
+
+      namespace internal
```

Comment:
i think this way those functions are exposed to users, hm... maybe move it right after `#ifndef DOXYGEN`, you already have `internal` namespaces somewhere there